### ISSUE TITLE
Antenna Type and channel bandwidth 

### ISSUE THREAD

**Issue by ines-s-santos** [intrig-unicamp/mininet-wifi#105]
Hi!

I have two important questions in order to proceed with my work:

1. Can I choose the type of antenna? For example, a tri-sectorized one (not omnidirectional).

2. Can I define a channel bandwidth? I want to use a 10 MHz channel, it is possible?

Thank you.

**Comment by ramonfontes** [comment]
Hi,

1. Not yet. It could be implemented with wmediumd.

2. Yes, you can. I just added support to ht_capab and you can use the param
with the same name when you create APs. For example:

*ap1 = net.addAccessPoint('ap1', ssid="simplewifi", mode="n",
ht_capab="[HT40-][SHORT-GI-20][SHORT-GI-40]", channel="5")*

Please refer to hostapd.conf
<https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf> for further
information on how to configure HT. If you need support to VHT please let
me know which parameters you need to make it work.

BTW, I don't know what is the impact of this type of configuration during
data transmission (I couldn't read about it yet). Any help is welcome ;).

2017-08-29 13:39 GMT-03:00 ines-s-santos <notifications@github.com>:

**Comment by ines-s-santos** [comment]
Alright thank you, I'll try that. Other question:
- And can I change the frequency ? I know that the frequency depends on the mode, but can I define a frequency value and not use the mode ?

**Comment by ramonfontes** [comment]
Yes, you can. If you change the channel, you change the frequency.

**Comment by ines-s-santos** [comment]
Yes, but I want a specific frequency, 2 GHz, and do not correspond to any channel frequency.

**Comment by ramonfontes** [comment]
Hi Inês,

Could you please let me know which wifi technology works with 2Ghz?

Sent from my android


On 31 Aug 2017 12:14, "ines-s-santos" <notifications@github.com> wrote:

Yes, but I want a specific frequency, 2 GHz, and do not correspond to any
channel frequency.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/intrig-unicamp/mininet-wifi/issues/105#issuecomment-326327589>,
or mute the thread

**Comment by ines-s-santos** [comment]
It is not a Wi-Fi technology, I am simulating a Mobile Network, LTE (4G).

**Comment by ramonfontes** [comment]
As probably you know, mac80211_hwsim was designed for wifi networks. If you
want to simulate 4G, you should manipulate bw, loss, latency and delay.

**Comment by ines-s-santos** [comment]
Alright. Those parameters are defined for each link ? Or can be defined for the network in general or for an access point ?

**Comment by ramonfontes** [comment]
For the network in general (by default). You can change it if you want.
https://github.com/intrig-unicamp/mininet-wifi/blob/master/mininet/wifiLink.py

**Comment by ines-s-santos** [comment]
Okay thank you ! 
I'll take the opportunity to ask another question:
I want to do a graph with the throughput depending on the distance with mobility, so I need to print the sta distance as she moves, so how can I get the distance of a specific sta at any second and print it to a txt?

**Comment by ramonfontes** [comment]
Since you have the position of every node (node.params['position']), you
just have to use the formula for calculating the distance.

**Comment by NadiaMOUAWAD** [comment]
Hello, I just read the conversation. 
I am wondering if it is possible to integrate 4G in mininet-wifi knowing that mac80211_hwsim was designed for wifi networks. If it is so, how can we do this? Could it be by simply changing the network parameters?

**Comment by ramonfontes** [comment]
Hello,

Yes, you can do this by changing the network parameters. A new mode (4G
mode) could be created for wmediumd as well. As a consequence, the module
mac80211_hwsim could be extended as well.

2017-11-01 10:22 GMT-03:00 NadiaMOUAWAD <notifications@github.com>:
